### PR TITLE
[IT-3697] Fix SSM access to docker SC product

### PR DESCRIPTION
**Pull request by jesusaurus**
Use the correct base image with SSM pre-installed, and ensure that the SSM user has sudo access.

### CODE REVIEW

File: sceptre/scipool/config/bmgfki/sc-product-ec2-linux-docker.yaml [hunk 1]
```diff
@@ -58,3 +58,7 @@ sceptre_user_data:
       Info:
         LoadTemplateFromURL: 'https://{{stack_group_config.admincentral_cf_bucket}}.s3.amazonaws.com/{{stack_group_config.service_catalog_library}}/v1.4.0/ec2/sc-ec2-linux-docker.yaml'
       Name: 'v1.4.0'
+    - Description: 'Update Amazon Linux AMI {{ range(1, 10000) | random }}'
```
Comment (zaro0508) [inline sceptre/scipool/config/bmgfki/sc-product-ec2-linux-docker.yaml:61]: need to remove `{{ range(1, 10000) | random }}` from previous version

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: sceptre/scipool/config/bmgfki/sc-product-ec2-linux-docker.yaml (hunk 1 above)
Comment (jesusaurus) [reply sceptre/scipool/config/bmgfki/sc-product-ec2-linux-docker.yaml:61]: thanks

File: sceptre/scipool/config/develop/sc-product-ec2-linux-docker.yaml [hunk 2]
```diff
@@ -59,5 +59,5 @@ sceptre_user_data:
       Name: 'v1.4.0'
     - Description: 'Update Amazon Linux AMI {{ range(1, 10000) | random }}'
       Info:
-        LoadTemplateFromURL: 'https://{{stack_group_config.admincentral_cf_bucket}}.s3.amazonaws.com/{{stack_group_config.service_catalog_library}}/master/ec2/sc-ec2-linux-docker.yaml'
+        LoadTemplateFromURL: 'https://{{stack_group_config.admincentral_cf_bucket}}.s3.amazonaws.com/{{stack_group_config.service_catalog_library}}/v1.4.3/ec2/sc-ec2-linux-docker.yaml'
```
Comment (zaro0508) [inline sceptre/scipool/config/develop/sc-product-ec2-linux-docker.yaml:62]: need to remove {{ range(1, 10000) | random }} from previous version as well.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: sceptre/scipool/config/develop/sc-product-ec2-linux-docker.yaml (hunk 2 above)
Comment (jesusaurus) [reply sceptre/scipool/config/develop/sc-product-ec2-linux-docker.yaml:62]: It was already removed in the dev account

**Pull request merged**